tests/test_enterprise/test_course_catalog_api.py

```diff
@@ -328,3 +328,18 @@ def get_course_run(*args, **kwargs):  # pylint: disable=unused-argument
 
         actual_result = self.api.get_common_course_modes(course_runs)
         assert actual_result == expected_result
+
+    @mock.patch('enterprise.course_catalog_api.course_discovery_api_client')
+    @ddt.data(
+        (23, 'course-id', {'courses': {}}, False),
+        (45, 'fancy-course', {'courses': {'fancy-course': True}}, True),
+        (93, 'my-course', {'courses': {'my-course': False}}, False)
+    )
+    @ddt.unpack
+    def test_course_in_catalog(self, catalog_id, course_id, api_resp, expected, mock_discovery_client_creator):
```

Comment:
@haikuginger Can we rename `mock_discovery_client_creator` to `mock_discovery_client_factory` (or do you think that would be confusing since `enterprise.course_catalog_api.course_discovery_api_client` is not a "normal" factory defined in test code?